cmd/flux/check_test.go

```diff
@@ -0,0 +1,40 @@
+// +build e2e
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+
+	"github.com/fluxcd/flux2/internal/utils"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+func TestCheckPre(t *testing.T) {
+	jsonOutput, err := utils.ExecKubectlCommand(context.TODO(), utils.ModeCapture, rootArgs.kubeconfig, rootArgs.kubecontext, "version", "--output", "json")
+	if err != nil {
+		t.Fatalf("Error running utils.ExecKubectlCommand: %v", err.Error())
+	}
+
+	var versions map[string]version.Info
+	if err := json.Unmarshal([]byte(jsonOutput), &versions); err != nil {
+		t.Fatalf("Error unmarshalling: %v", err.Error())
+	}
+
+	kubectlVersion := strings.TrimPrefix(versions["clientVersion"].GitVersion, "v")
+	k8sVersion := strings.TrimPrefix(versions["serverVersion"].GitVersion, "v")
+
+	cmd := cmdTestCase{
+		args:            "check --pre",
+		wantError:       false,
+		testClusterMode: ExistingClusterMode,
+		templateValues: map[string]string{
+			"kubectlVersion": kubectlVersion,
```

Comment:
Let's rename these to `clientVersion` and  `serverVersion` same in the goldfile please.